### PR TITLE
added functionality for saving pdf and pxrd plots and offset pdf plots

### PR DESCRIPTION
**Pull request by harpaf13**

### CODE REVIEW

File: matador/plotting/convergence_plotting.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
 from collections import defaultdict
 from traceback import print_exc
 import numpy as np
+import matplotlib
+matplotlib.use('TkAgg')
```
Comment (ml-evs) [inline matador/plotting/convergence_plotting.py]: Did you mean to include the changes in this file?

File: matador/plotting/pdf_plotting.py [hunk 2]
```diff
@@ -36,22 +36,39 @@ def plot_pdf(pdf, other_pdfs=None, labels=None):
...
+        ax1.set_xlim(1,maxr)
     if other_pdfs is not None:
         for ind, _pdf in enumerate(other_pdfs):
             if isinstance(_pdf, Crystal):
                 _pdf = _pdf.pdf
             elif isinstance(_pdf, dict) and 'pdf' in _pdf:
                 _pdf = _pdf['pdf']
             if isinstance(_pdf, PDF):
-                ax1.plot(_pdf.r_space, _pdf.gr, label=labels[ind+1] if labels else _pdf.label, ls='--', alpha=1)
+                ax1.plot(_pdf.r_space, _pdf.gr+20*(ind+1), label=labels[ind+1] if labels else _pdf.label, ls='--', alpha=1)
```
Comment (ml-evs) [inline matador/plotting/pdf_plotting.py]: Can you add this offset as a parameter instead please? (with 0 by default)

File: matador/plotting/pxrd_plotting.py [hunk 3]
```diff
@@ -32,13 +32,13 @@ def plot_pxrd(pxrds, two_theta_range=(10, 70), figsize=None):
...
         height = max(0.5, 5/len(pxrds))
-        figsize = (8, height)
+        figsize = (10, height)
     import matplotlib.pyplot as plt
     fig = plt.figure(figsize=figsize)
     for ind, pxrd in enumerate(pxrds):
         ax = fig.add_subplot(111)
         ax.plot(pxrd.two_thetas, 0.9*pxrd.spectrum + ind)
-        ax.text(0.95, ind+0.1, '{} ({})'.format(pxrd.formula, pxrd.spg),
+        ax.text(0.95, ind+0.15, '{} ({})'.format(pxrd.formula, pxrd.spg),
```
Comment (ml-evs) [inline matador/plotting/pxrd_plotting.py]: and the same thing here

File: matador/plotting/pxrd_plotting.py [hunk 4]
```diff
@@ -32,13 +32,13 @@ def plot_pxrd(pxrds, two_theta_range=(10, 70), figsize=None):
         pxrds = [pxrds]
     if figsize is None:
         height = max(0.5, 5/len(pxrds))
-        figsize = (8, height)
+        figsize = (10, height)
```
Comment (ml-evs) [inline matador/plotting/pxrd_plotting.py]: Can you leave the default and just pass your own figsize when you call it?

File: matador/plotting/convergence_plotting.py (hunk 1 above)
Comment (harpaf13) [reply matador/plotting/convergence_plotting.py]: no

File: matador/plotting/pxrd_plotting.py (hunk 4 above)
Comment (harpaf13) [reply matador/plotting/pxrd_plotting.py]: defaults for other figs are 10 though but yes can do

**Comment by harpaf13** [comment]
All comments addressed in recent pushes

**Comment by ml-evs** [comment]
> All comments addressed in recent pushes

Linting is still failing and it still looks like this is about 50 commits behind develop, can you rebase please?

File: matador/plotting/convergence_plotting.py [hunk 5]
```diff
@@ -76,6 +76,7 @@ def plot_cutoff_kpt_grid(data, forces=False, max_energy=50, legend=True, colour_
             ax=ax[3],
             reference='last'
         )
+    plt.show()
```
Comment (ml-evs) [inline matador/plotting/convergence_plotting.py]: this still needs to be removed

Comment (harpaf13) [reply matador/plotting/convergence_plotting.py]: removed

File: matador/plotting/pdf_plotting.py [hunk 6]
```diff
@@ -36,22 +36,40 @@ def plot_pdf(pdf, other_pdfs=None, labels=None):
...
     if other_pdfs is not None:
         for ind, _pdf in enumerate(other_pdfs):
+            gr_offset = offset*(ind+1)
             if isinstance(_pdf, Crystal):
                 _pdf = _pdf.pdf
             elif isinstance(_pdf, dict) and 'pdf' in _pdf:
                 _pdf = _pdf['pdf']
             if isinstance(_pdf, PDF):
-                ax1.plot(_pdf.r_space, _pdf.gr, label=labels[ind+1] if labels else _pdf.label, ls='--', alpha=1)
+                ax1.plot(_pdf.r_space, _pdf.gr+offset, label=labels[ind+1] if labels else _pdf.label, ls='--', alpha=1)
```
Comment (ml-evs) [inline matador/plotting/pdf_plotting.py]:
```suggestion
                ax1.plot(_pdf.r_space, _pdf.gr + (ind+1) * offset, label=labels[ind+1] if labels else _pdf.label, ls='--', alpha=1)
```

File: matador/plotting/pdf_plotting.py [hunk 7]
```diff
@@ -36,22 +36,40 @@ def plot_pdf(pdf, other_pdfs=None, labels=None):
...
             elif isinstance(_pdf, tuple):
-                ax1.plot(_pdf[0], _pdf[1], alpha=1, labels=labels[ind+1] if labels else None, ls='--')
+                ax1.plot(_pdf[0], _pdf[1]+offset, alpha=1, labels=labels[ind+1] if labels else None, ls='--')
```
Comment (ml-evs) [inline matador/plotting/pdf_plotting.py]:
```suggestion
                ax1.plot(_pdf[0], _pdf[1] + (ind+1) * offset, alpha=1, labels=labels[ind+1] if labels else None, ls='--')
```

File: matador/plotting/pdf_plotting.py [hunk 8]
```diff
@@ -36,22 +36,40 @@ def plot_pdf(pdf, other_pdfs=None, labels=None):
...
     ax1.set_ylabel('Pair distribution function, $g(r)$')
-    ax1.set_xlim(0, pdf.rmax)
+    ax1.get_yaxis().set_ticks([])
+    if maxr is None:
+        ax1.set_xlim(1, pdf.rmax)
+    else:
+        ax1.set_xlim(1,maxr)
     if other_pdfs is not None:
         for ind, _pdf in enumerate(other_pdfs):
+            gr_offset = offset*(ind+1)
```
Comment (ml-evs) [inline matador/plotting/pdf_plotting.py]:
```suggestion

```

**Pull request merged**